docs/cache.txt

```diff
@@ -0,0 +1,234 @@
+   :language: php
+   :emphasize-lines: 11,12
+   :dedent:
+
+Then run the migration:
+
+.. code-block:: none
+
+   $ php artisan migrate
+
+Alternatively, the index can be created using :mdb-shell:`MongoDB Shell <>` (``mongosh``):
+
+.. code-block:: ts
+
+   db.cache.createIndex(
+     /* Field that holds the expiration date */
+     { expires_at: 1 },
+     /* Delay to remove items after expiration */
+     { expireAfterSeconds: 0 }
+   )
+
+If you use Locks, disable ``lock_lottery`` by setting the probability to ``0``:
+
+.. code-block:: php
+   :highlightLines: [4]
+
+   'stores' => [
+       'mongodb' => [
+           'driver' => 'mongodb',
+           'connection' => 'mongodb',
+           'lock_lottery' => [0, 100], // Disabled
+       ],
+   ],
+
+Using MongoDB cache
+-------------------
+
+The Laravel cache can be used to store any serializable data using the facade
+``Illuminate\Support\Facades\Cache``:
+
+In this example:
+- Gets the cache repository with the store ``mongodb``,
+- Tries to read and return the cache item named ``foo``,
+- If missing, calls the closure to compute the value, store this value forever and return it.
+
+..code-block:: php
+
+   use Illuminate\Support\Facades\Cache;
+
+   $value = Cache::store('mongodb')->get('foo', function () {
+       return [1, 2, 3];
+   });
+
+The default, cached objects do not expire. It is possible to define an expiry time.
+
+..code-block:: php
+
+   Cache::store('mongodb')->set('foo', 'abc', '1 day');
+
+Incrementing and decrementing a value is also supported, the value must
+be initialized before. The following example initialize the counter to ``3``,
+adds 5 and removes 2.
+
+..code-block:: php
+
+   Cache::store('mongodb')->set('counter', 3);
+   Cache::store('mongodb')->increment('counter', 5);
+   Cache::store('mongodb')->decrement('counter', 2);
+
+.. note::
+
+   {+odm-short+} supports incrementing and decrementing with integer and float values.
+
+Configuring MongoDB as default cache
+------------------------------------
+
+To use the ``mongodb`` store by default, change the default store in
+``config/cache.php``:
+
+.. code-block:: php
+   :emphasize-lines: 2
+
+   return [
+       'default' => env('CACHE_STORE', 'mongodb'),
+
+       'stores' => [
+           'mongodb' => [
+               'driver' => 'mongodb',
+               'connection' => 'mongodb',
+           ],
+       ],
+   ];
+
+The variable ``CACHE_STORE`` could be set in your environment or in
+the ``.env`` file. Update or remove it:
+
+.. code-block:: none
+
+   CACHE_STORE=mongodb
+
+Then you can use the ``Illuminate\Support\Facades\Cache`` facade and
+automatic injection:
+
+.. code-block:: php
+
+   use Illuminate\Support\Facades\Cache;
+
+   Cache::get('foo', 5);
+
+This example shows how to use automatic injection of the cache
+manager, using the default store. It is a controller that
+increments a counter each time it is invoked.
+
+
+.. code-block:: php
+   :emphasize-lines: 9,14
```

Comment:
I think you may have meant to highlight 9 and 13? The function names?

```suggestion
   :emphasize-lines: 9,13
```

Reply:
I think 14 is correct, it's the line with `return $this->cache->increment('counter');`

Reply:
You are right. Both lines need to be udpated.
```suggestion
   :emphasize-lines: 8,13
```

Reply:
I think these should be lines 10 and 15 to highlight the `CacheManager` line and the `return this-> ...` line